ocaml/xapi/nm.ml

```diff
@@ -618,6 +597,37 @@ let bring_pif_up ~__context ?(management_interface = false) (pif : API.ref_PIF)
                     in
                     (conf, gateway)
               in
+              let static =
+                match address_type with
+                | `IPv4 ->
+                    rc.API.pIF_ip_configuration_mode = `Static
+                | `IPv6 ->
+                    rc.API.pIF_ipv6_configuration_mode = `Static
+              in
+              let dns =
+                if static then
+                  if rc.API.pIF_DNS <> "" then
+                    let nameservers =
+                      List.map Unix.inet_addr_of_string
+                        (String.split ',' rc.API.pIF_DNS)
+                    in
+                    let domains =
+                      match List.assoc_opt "domain" rc.API.pIF_other_config with
+                      | None ->
+                          []
+                      | Some domains -> (
+                        try String.split ',' domains
+                        with _ ->
+                          warn "Invalid DNS search domains: %s" domains ;
+                          []
+                      )
+                    in
+                    (nameservers, domains)
+                  else
+                    ([], [])
+                else
+                  ([], [])
```

Comment:
A match would make this slightly neater

```suggestion
                match static, rc.API.pIF_DNS with
                | false, _ | true, "" -> ([], [])
                | true, pif_dns ->
                    let nameservers =
                      List.map Unix.inet_addr_of_string
                        (String.split ',' pif_dns)
                    in
                    let domains =
                      match List.assoc_opt "domain" rc.API.pIF_other_config with
                      | None ->
                          []
                      | Some domains -> (
                        try String.split ',' domains
                        with _ ->
                          warn "Invalid DNS search domains: %s" domains ;
                          []
                      )
                    in
                    (nameservers, domains)
```

Reply:
I don't think `XStringext.String.split` can raise exceptions, the try with seems useless here. It would be eveen better to replace it with a `String.split_on_char`

What would be invalid DNS domains? if the list is empty or there are empty strings in it?

Reply:
Since we take it from the PIF record, I don't see how there can be an invalid entries since it's tested when set on the PIF.